docs/root/start/quick-start/securing.rst

```diff
@@ -0,0 +1,198 @@
+   :emphasize-lines: 6-9
+   :caption: :download:`envoy-demo-tls-validation.yaml <_include/envoy-demo-tls-validation.yaml>`
+
+You can also ensure that the "Subject Alternative Names" for the cerficate match.
+
+This is commonly used by web certificates (X.509) to identify the domain or domains that a
+certificate is valid for.
+
+.. literalinclude:: _include/envoy-demo-tls-validation.yaml
+   :language: yaml
+   :linenos:
+   :lineno-start: 43
+   :lines: 43-53
+   :emphasize-lines: 6-7, 10-11
+   :caption: :download:`envoy-demo-tls-validation.yaml <_include/envoy-demo-tls-validation.yaml>`
+
+.. note::
+
+   If the "Subject Alternative Names" for a certificate are for a wildcard domain, eg ``*.example.com``,
+   this is what you should use when matching with ``match_subject_alt_names``.
+
+.. note::
+
+   See :ref:`here <envoy_v3_api_msg_extensions.transport_sockets.tls.v3.CertificateValidationContext>` to view all
+   of the possible configurations for certificate validation.
+
+.. _start_quick_start_securing_sni:
+
+Secure an endpoint with SNI
+---------------------------
+
+``SNI`` is an extension to the ``TLS`` protocol through which a client indicates the hostname of the server
+that they are connecting to during the ``TLS`` negotiation.
+
+To enforce ``SNI`` on a listening connection, you should set the
+:ref:`filter_chain_match <envoy_v3_api_msg_config.listener.v3.FilterChainMatch>` of the
+:ref:`listener <envoy_v3_api_msg_config.listener.v3.Listener>`:
+
+.. literalinclude:: _include/envoy-demo-tls-sni.yaml
+   :language: yaml
+   :linenos:
+   :lineno-start: 27
+   :lines: 27-35
+   :emphasize-lines: 2-4
+   :caption: :download:`envoy-demo-tls-sni.yaml <_include/envoy-demo-tls-sni.yaml>`
+
+.. _start_quick_start_securing_sni_client:
+
+Connect to an endpoint securely with SNI
+----------------------------------------
+
+When connecting to a ``TLS`` endpoint that is protected by ``SNI`` you can set
+:ref:`sni <envoy_v3_api_field_extensions.transport_sockets.tls.v3.UpstreamTlsContext.sni>` in the configuration
+of the :ref:`UpstreamTLSContext <envoy_v3_api_msg_extensions.transport_sockets.tls.v3.UpstreamTlsContext>`.
+
+.. literalinclude:: _include/envoy-demo-tls-sni.yaml
+   :language: yaml
+   :linenos:
+   :lineno-start: 56
+   :lines: 56-61
+   :emphasize-lines: 6
+   :caption: :download:`envoy-demo-tls-sni.yaml <_include/envoy-demo-tls-sni.yaml>`
+
+When connecting to an Envoy endpoint that is protected by ``SNI``, this must match one of the
+:ref:`server_names <envoy_v3_api_field_config.listener.v3.FilterChainMatch.server_names>` set in the endpoint's
+:ref:`filter_chain_match <envoy_v3_api_msg_config.listener.v3.FilterChainMatch>`, as
+:ref:`described above <start_quick_start_securing_sni>`.
+
+.. note::
+
+   When connecting to a ``TLS`` endpoint that is not protected by ``SNI``, this configuration is ignored, so it is
+   generally advisable to always set this to the DNS name of the endpoint you are connecting to.
```

Comment:
not 100% sure if this is correct advice, but  seems to make sense to me

Reply:
See above about SNI.